sdks/python/apache_beam/io/gcp/experimental/spannerio.py

```diff
@@ -1247,14 +1253,17 @@ def process(self, element):
           else:
             raise ValueError("Unknown operation action: %s" % m.operation)
           batch_func(**m.kwargs)
-
-      self.service_metric.call('ok')
     except (ClientError, GoogleAPICallError) as e:
-      self.service_metric.call(str(e.code.value))
+      for service_metric in self.service_metrics.values():
+        service_metric.call(str(e.code.value))
```

Comment:
Should we only be updating metrics for the tables that we tried to update here ?

Reply:
This should do exactly that since an error fails the whole batch no matter what kind of error it is.